pkg/pushsync/pushsync.go

```diff
@@ -351,6 +351,7 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 
 	resultChan := make(chan receiptResult, 1)
 	doneChan := make(chan struct{})
+	defer close(doneChan)
```

Comment:
you cannot do this. please see https://go.dev/play/p/YO9nL9sKMuy
the dangling goroutines would panic

Reply:
this is the done channel, it's read only.

Reply:
oh sorry, looked at the wrong line. so we have a deadlock on the testnet without this, right?

Reply:
yeah, if pushToClosest returns with error, there is a goroutine leak.